### PR TITLE
Feature edit requisitions

### LINKED ISSUE #5660: Allow users to edit requisitions
We should allow users to edit requisitions that are not yet delivered.  Requisitions seem to be an excellent way to allow users to experiment with creating orders to be filled later.  However, if the user makes a mistake, they need to cancel the requisition entirely and begin again.  This isn't a nice user experience.

My understanding of requisitions is that they do not have any dependencies - you can create one at any time, request from any depot, no matter how much stock is or isn't available in the depot(s).  If that is true, we should be able to easily allow users to edit requisitions with minimal risk.

### PR DESCRIPTION
**Pull request by lomamech**
- Allow users to edit requisitions
- Add E2E test for Update requisition

closes #5660

### CODE REVIEW

**Review by mbayopanda — COMMENTED** [review]
The feature works well with requisition for depot, but I just noticed that for service, the concerned service field is not defined

image

**Comment by lomamech** [comment]
> The feature works well with requisition for depot, but I just noticed that for service, the concerned service field is not defined
> 
> image

**Comment by lomamech** [comment]
@mbayopanda it's done

File: server/controllers/stock/requisition/requisition.js [hunk 1]
```diff
@@ -280,7 +285,7 @@ exports.update = async (req, res, next) => {
 
     if (requisitionItems && requisitionItems.length) {
       transaction
-        .addQuery('DELETE FROM stock_requisition_item WHERE requisition_uuid = ?;', [binarize(requisition), uuid]);
+        .addQuery('DELETE FROM stock_requisition_item WHERE requisition_uuid = ?;', [uuid]);
```
Comment (mbayopanda) [inline server/controllers/stock/requisition/requisition.js:288]: 👍🏽

File: client/src/js/components/bhServiceOrDepot/bhServiceOrDepot.html [hunk 2]
```diff
@@ -31,6 +31,7 @@
   <!-- service  -->
   <div ng-if="$ctrl.requestorType.type_key === 'service'">
     <bh-service-select
+      service-uuid="$ctrl.uuid"
```
Comment (mbayopanda) [inline client/src/js/components/bhServiceOrDepot/bhServiceOrDepot.html:34]: 👍🏽

**Review by mbayopanda — APPROVED** [review]
LGTM 👍🏽

**Comment by jniles** [comment]
LGTM as well.

bors r+

**Pull request merged**